lib/primitives/address.js

```diff
@@ -587,9 +656,13 @@ class Address {
     } else {
       assert(type === Address.types.WITNESS, 'Wrong version (non-witness).');
       assert(version >= 0 && version <= 16, 'Bad program version.');
-      if (version === 0 && type === Address.types.WITNESS) {
-        assert(hash.length === 20 || hash.length === 32,
-          'Witness program hash is the wrong size.');
+      if (type === Address.types.WITNESS) {
+        if (version === 0)
+          assert(hash.length === 20 || hash.length === 32,
+            'Witness program hash is the wrong size.');
+        if (version === 1)
+          assert(hash.length === 32,
+            'taproot program hash is the wrong size.');
```

Comment:
Readability and nested-if clarity (https://blog.codecentric.de/en/2014/02/curly-braces/ 😱 )
Also, correct terminology

```suggestion
      if (type === Address.types.WITNESS) {
        if (version === 0) {
          assert(hash.length === 20 || hash.length === 32,
            'Witness version 0 program hash is the wrong size.');
        }
        if (version === 1) {
          assert(hash.length === 32,
            'Witness version 1 program hash is the wrong size.');
        }
```